### PR TITLE
tree-wide: introduce strextendf_with_separator() and use strextend() or freinds

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by yuwata** [comment]
@poettering Could you review this? Especially, the first commit.

File: src/network/networkd-route.c [hunk 1]
```diff
@@ -773,13 +773,17 @@ static void log_route_debug(const Route *route, const char *str, const Link *lin
...
+                        _cleanup_free_ char *buf = NULL;
                         union in_addr_union a = m->gateway.address;
 
                         (void) in_addr_to_string(m->gateway.family, &a, &buf);
-                        joined = strjoin(gw_alloc, gw_alloc ? "," : "", strna(buf), m->ifname ? "@" : "", strempty(m->ifname));
-                        if (joined)
-                                free_and_replace(gw_alloc, joined);
+                        (void) strextend_with_separator(&gw_alloc, ",", strna(buf));
+                        if (m->ifname)
+                                (void) strextend_with_separator(&gw_alloc, "@", m->ifname);
```
Comment (poettering) [inline src/network/networkd-route.c]: hmm, use regular `strextend()` here?

File: src/network/networkd-route.c [hunk 2]
```diff
@@ -773,13 +773,17 @@ static void log_route_debug(const Route *route, const char *str, const Link *lin
...
+                        else if (m->ifindex > 0)
+                                (void) strextendf_with_separator(&gw_alloc, "@", "%"PRIu32, m->ifindex);
```
Comment (poettering) [inline src/network/networkd-route.c]:
```c
(void) strextendf(&gw_alloc, "@%" PRIu32, m->ifindex);        
```

would be less surprsiing, no?

File: src/network/networkd-route.c [hunk 3]
```diff
@@ -773,13 +773,17 @@ static void log_route_debug(const Route *route, const char *str, const Link *lin
...
+                        /* See comments in config_parse_multipath_route(). */
+                        (void) strextendf_with_separator(&gw_alloc, ":", "%"PRIu32, m->weight + 1);
```
Comment (poettering) [inline src/network/networkd-route.c]:
```
(void) strextendf(&gw_alloc, ":%"PRIu32, m->weight + 1);
```

i.e. the special logic that the _with_separator() functions have is that they suppress the separator at the beginning of the string. But that's not necessary here, we already appeneded the address anyway, here we just append a bit more. Hence, I think it's a bit confusing here if we'd use the special logic here even though we know for sure it's not going to be used..

**Review by poettering — CHANGES_REQUESTED** [review]
these are *nice* clean-ups.

**Comment by yuwata** [comment]
@poettering Thank you for the review. The above points are addressed. PTAL.

**Comment by mrc0mmand** [comment]
```
+ env --unset=UNIFIED_CGROUP_HIERARCHY --unset=SYSTEMD_NSPAWN_UNIFIED_HIERARCHY timeout --foreground 1200 /systemd-meson-build/systemd-nspawn --register=no --kill-signal=SIGKILL --directory=/var/tmp/TEST-01-BASIC_sanitizers-nspawn/unprivileged-nspawn-root --setenv=SYSTEMD_UNIT_PATH=/usr/lib/systemd/tests/testdata/testsuite-01.units:/usr/lib/systemd/tests/testdata/units: -U --private-network /usr/lib/systemd/systemd-under-asan systemd.unit=testsuite.target systemd.wants=testsuite-01.service systemd.wants=end.service
Spawning container unprivileged-nspawn-root on /var/tmp/TEST-01-BASIC_sanitizers-nspawn/unprivileged-nspawn-root.
Press ^] three times within 1s to kill container.
Selected user namespace base 970129408 and range 65536.
Failed to mount sysfs (type sysfs) on /sys/full (1|2|4|8 ""): No such file or directory
Failed to allocate scope: Interrupted system call
E: nspawn failed with exit code 1
```
:thinking:

**Pull request merged**